src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,454 @@
+                        case "Double":
+                            entry = new models.doubleEntry(rawEntry);
+                            break;
+                        case "Boolean":
+                            entry = new models.booleanEntry(rawEntry);
+                            break;
+                        case "Enum":
+                            entry = new models.enumEntry(rawEntry);
+                            break;
+                        case "Time":
+                            entry = new models.timeEntry(rawEntry);
+                            break;
+                        case "Size":
+                            entry = new models.sizeEntry(rawEntry);
+                            break;
+                    }
+
+                    entry.init();
+                    return entry;
+                });
+
+                this.allEntries(_.sortBy(settingsEntries, x => x.keyName()));
+            });
+    }
+
+    private isEntriesValid() {
+        let valid = true;
+        let firstError = true;
+
+        this.allEntries().map(entry => {
+            if (this.getEntryStateClass(entry) === "customized-item-with-error") {
+                valid = false;
+
+                if (firstError) {
+                    firstError = false;
+                    this.setCategory(entry.data.Metadata.Category);
+                }
+            }
+        });
+
+        return valid;
+    }
+    
+    save() {
+        if (!this.isEntriesValid()) {
+            if (this.viewMode() === "summaryMode") {
+                this.switchToEditMode()
+            }
+
+            return;
+        }
+
+        const settingsToSave = this.allEntries().map(entry => {
+            if (entry instanceof models.databaseEntry) {
+                return entry.getEntrySetting();
+            }
+        }).filter(x => !!x);
+
+        const settingsToSaveSorted = _.sortBy(settingsToSave, x => x.key);
+        
+        const saveSettingsModel = new saveDatabaseSettingsConfirm((settingsToSaveSorted))
+
+        saveSettingsModel.result.done(result => {
+            if (result.can) {
+                this.spinners.save(true);
+
+                new saveDatabaseSettingsCommand(this.activeDatabase(), settingsToSaveSorted)
+                    .execute()
+                    .done(() => {
+                        this.fetchDatabaseSettings(this.activeDatabase())
+                            .done(() => this.processAfterFetch())
+                    })
+                   .always(() => this.spinners.save(false))
+                }
+            });
+
+        app.showBootstrapDialog(saveSettingsModel);
+    }
+
+    refresh() {
+        this.confirmRefresh()
+            .then(result => {
+                if (result.can) {
+                    this.spinners.refresh(true);
+
+                    this.fetchDatabaseSettings(this.activeDatabase(), true)
+                        .done(() => {
+                            this.processAfterFetch();
```

Comment:
can we move processAfterFetch to fetchDatabaseSettings?

Reply:
Prefer not to because the first time `processAfterFetch `is executed is in compositionComplete, and `fetchDatabaseSettings `is in the canActivate